modules/fedimint-ln/src/db.rs

```diff
@@ -118,3 +118,143 @@ impl_db_lookup!(
     key = LightningGatewayKey,
     query_prefix = LightningGatewayKeyPrefix
 );
+
+#[cfg(test)]
+mod fedimint_migration_tests {
+    use std::collections::BTreeMap;
+    use std::env;
+    use std::path::Path;
+
+    use fedimint_core::core::LEGACY_HARDCODED_INSTANCE_ID_LN;
+    use fedimint_core::db::apply_migrations;
+    use fedimint_core::module::DynModuleGen;
+    use fedimint_testing::apply_to_databases;
+    use futures::StreamExt;
+    use strum::IntoEnumIterator;
+
+    use crate::db::{
+        AgreedDecryptionShareKeyPrefix, ContractKeyPrefix, ContractUpdateKeyPrefix, DbKeyPrefix,
+        LightningGatewayKeyPrefix, ProposeDecryptionShareKeyPrefix,
+    };
+    use crate::LightningGen;
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn verify_lightning_database_migration() {
+        // Only run the database migration test if the database backup directory
+        // environment variable has been defined
+        let backup_dir = env::var("FM_TEST_DB_BACKUP_DIR");
+        if let Ok(backup_dir) = backup_dir {
+            let mut migrated_values = BTreeMap::new();
+            apply_to_databases(
```

Comment:
`apply_to_databases` doesn't seem informative. I'm confused what's going on, and will have to read impl of `apply_to_databases` first.

Reply:
Removed